src/components/ProposalTable.vue

```diff
@@ -0,0 +1,242 @@
+<template>
+	<div class="dao_table_container">
+		<a-table
+			:dataSource="dataSource"
+			:columns="columns"
+			bordered
+			:pagination="false"
+		>
+			<template #title>
+				<a-col>
+					<h3 style="text-align: center">Available Proposals</h3>
+				</a-col>
+			</template>
+			<template #bodyCell="{ record, column }">
+				<template v-if="column.key === 'name'">
+					<a :href="'//' + record.url" target="_blank">
+						{{ record.name }}
+					</a>
+				</template>
+				<template v-if="column.key === 'voting_start'">
+					{{
+						secToFormat(record.voting_start, "lll") +
+						" - " +
+						secToFormat(record.voting_end, "lll")
+					}}
+				</template>
+				<template v-if="column.key === 'type'">
+					{{ ProposalType[record.type] }}
+				</template>
+			</template>
+		</a-table>
+	</div>
+	<a-col :offset="15">
+		<a-pagination
+			:hideOnSinglePage="true"
+			v-model:current="currentPage"
+			:pageSize="ROWS_PER_PAGE"
+			:total="totalDataRowsCount"
+		>
+			<template #itemRender="{ type, originalElement }">
+				<a
+					v-if="type === 'prev'"
+					@click="handlePagination(PaginationCallType.NAV_PREV)"
+					><left-outlined
+				/></a>
+				<a
+					v-else-if="type === 'next'"
+					@click="handlePagination(PaginationCallType.NAV_NEXT)"
+					><right-outlined
+				/></a>
+				<component
+					@click="
+						handlePagination(
+							PaginationCallType.JUMP_PAGE,
+							originalElement.children[0].children
+						)
+					"
+					:is="originalElement"
+					v-else
+				></component>
+			</template>
+		</a-pagination>
+	</a-col>
+</template>
+
+<script lang="ts">
+import {
+	VALIDATE_MESSAGES,
+	openErrorNotificationWithIcon,
+	UNSUCCESSFUL,
+	ROWS_PER_PAGE,
+	ProposalType,
+} from "@/constants";
+import { ProposalTableData, PaginationCallType } from "@/types";
+import { defineComponent, reactive, ref, toRefs } from "vue";
+import DaoID from "../store/DaoID";
+import ProposalTableStore from "../store/ProposalTableStore";
+import { secToFormat } from "../utility";
+import { decodeProposalParams } from "@/indexer";
+import {
+	executeReq,
+	searchProposalsByAppIdReq,
+	getProposalCursorReq,
+} from "@/api";
+
+export default defineComponent({
+	name: "ProposalTable",
+	data() {
+		return {
+			key: "ProposalKey",
+			columns: [
+				{
+					title: "Proposal Name",
+					key: "name",
+					dataIndex: "name",
```

Comment:
I think we have `PROPOSAL_LOCAL_STATE_MAP_KEY` with these values, we should update to use the constants then

Reply:
Done.